services/datamanager/builtin/builtin.go

```diff
@@ -655,9 +639,53 @@ func (svc *builtIn) Reconfigure(
 			svc.captureDir, deleteEveryNthValue, svc.syncer, svc.logger)
 	}
 
+	g.Success()
 	return nil
 }
 
+func (svc *builtIn) asyncInitDataSync() {
+	for goutils.SelectContextOrWait(svc.closedCtx, time.Second) {
+		svc.lock.Lock()
+		if svc.syncConfigUpdated {
```

Comment:
when `svc.syncConfigUpdated == true` we know that a previous `Reconfigure`  has occured which changed datasync related config, which requires us to call initSyncer and or closSyncer.

Once we have processed that config this goroutine sets `svc.syncConfigUpdated` to false.

We assume that we don't need to successfully process ALL datasync related configurations, we just need to attempt to process the current one until either Close() is called on data manager, we determine this is a local config managed robot (which doesn't support datasync), or datasync is initialized with the new config.